kuma/javascript/src/article.jsx

```diff
@@ -108,11 +109,7 @@ function highlightSections(article) {
 // kuma/static/js/components/local-anchor.js
 function addAnchors(article) {
     for (let heading of article.querySelectorAll('h2[id], h3[id]')) {
-        let anchor = document.createElement('a');
-        anchor.href = `#${heading.id}`;
-        anchor.classList.add('sectionLink');
```

Comment:
[See line 41 in this file.](https://github.com/mozilla/kuma/blob/b6e10cf7607a41b2fad830dbb77cea50502ae9cd/kuma/javascript/src/article.jsx#L41)

It's kinda sad what's going on here. The whole reason for using css-in-js (`emotion` in our case) is that when you remove the JS it should imply the removal of the CSS. But that's not the case here. It's because we're using `DOMElement.classList(string)` and not `css={styles.sectionLink}`. Gross.

I wonder, do we really need React for any of this stuff?? From what I can see, the `local-anchor.js` is pure/vanilla JS. Its only flaw is that it's not callable on repeats. You could easily solve that by changing that file to:

```javascript
window.prependLocalAnchors = (function() {
    'use strict';

    function prependLocalAnchors() {
        /// ...the magic....
    }

    // `Array.from` is not supported in IE, progressively enhance
    if (typeof Array.from !== 'undefined') {
        window.addEventListener('load', function() {
            prependLocalAnchors();
        });
    }
    return prependLocalAnchors;
})();
```

Then, `article.jsx` would just look something like this:
```javascript
function addAnchors(article) {
    // SOME GOOD COMMENT EXPLANATION
    if (window.prependLocalAnchors) {
        window.prependLocalAnchors();
    }
}
```

I don't know what that means for django-pipeline etc to share a piece of JS code across both worlds but it would certainly solve a lot of things. Also, `local-anchor.js` seems to depend on `window.mdnIcons` which I don't know if we have or want to have in React land.

Reply:
That should work. If you feel that is a better approach, we could go that route.